src/main/resources/templates/newOrder.html

```diff
@@ -15,7 +15,7 @@
 
     <p th:text="${order}" />
 
-    <form action="#" th:action="@{/saveOrder}" th:object="${order}" method="GET">
+    <form action="#" th:action="@{/saveOrder}" th:object="${Order}" method="GET">
```

Comment:
Tried to get your Thymeleaf annotations to work, but failed.

Reply:
Is it necessary to commit and push something that didn't fix an issue?  That may cause more issues.  
We typically use lowerCamelCase when naming variables, so order looks to be correct.